homeassistant/components/statistics/sensor.py

```diff
@@ -411,6 +412,20 @@ def _async_stats_sensor_state_listener(
         if not self._preview_callback:
             self.async_write_ha_state()
 
+    @callback
+    def _async_stats_sensor_state_change_listener(
+        self,
+        event: Event[EventStateChangedData],
+    ) -> None:
+        self._async_handle_new_state(event.data["new_state"])
+
+    @callback
+    def _async_stats_sensor_state_report_listener(
+        self,
+        event: Event[EventStateReportedData],
+    ) -> None:
+        self._async_handle_new_state(event.data["new_state"])
```

Comment:
Is `state_change` still needed if `state_reported` is listened to?

Reply:
I copied the mechanism from the Riemann integral thing, where they did the same.

Actually, I also tried out removing the update events already. My assumption was that every state report would be coming together with a state updated event, but that doesn't seem to be the case.

Reply:
With reported you don't have an`old_state` but instead you have an `old_last_reported` datetime. The `new_state` should be there for both.
I don't think we need neither `old_state` nor the `old_last_reported` so it should be sufficient to only work with `reported`?

Reply:
@gjohansson-ST: I guess you are talking about which attribute to use as the timestamp, correct? That had to be switched from last_updated to last_reported. Now last_reported is used for both types of event, and that is the only thing we need. But we definitely need both events, since changing values don't trigger the report events.

Reply:
@gjohansson-ST when subscribing to `state_reported` events, the listener *only* gets called when an existing state is unchanged. It's necessary to also subscribte to `state_changed` events. I think we should, in a separate PR, improve the docstrings for `async_track_state_report_event` and `EVENT_STATE_REPORTED`.